lib/brains/tests/test_subscription.py

```diff
@@ -98,7 +97,56 @@ def test_braintree_fails(self):
         eq_(self.braintree_error(res.json, 'payment_method_token'), ['91903'])
 
 
-class TestSubscriptionViewSet(APITest):
+class TestSubscriptionCancel(BraintreeTest):
+    gateways = {'sub': SubscriptionGateway}
+
+    def setUp(self):
+        super(TestSubscriptionCancel, self).setUp()
+        self.url = reverse('braintree:subscription.cancel')
+
+    def test_allowed(self):
+        self.allowed_verbs(self.url, ['post'])
+
+    def create_subscription(self):
+        method, product = create_method_all()
+        return BraintreeSubscription.objects.create(
+            paymethod=method, seller_product=product)
+
+    def post(self, obj, **kw):
+        data = {'subscription': obj.get_uri()}
+        data.update(**kw)
+        return self.client.post(self.url, data=data)
+
+    def test_post(self):
+        self.mocks['sub'].cancel.return_value = successful_subscription()
+
+        obj = self.create_subscription()
+        res = self.post(obj)
+        eq_(res.status_code, 200, res.status_code)
+
+        self.mocks['sub'].cancel.assert_called_with(obj.provider_id)
+
+    def test_fails_post(self):
+        self.mocks['sub'].cancel.return_value = error()
+
+        obj = self.create_subscription()
+        res = self.post(obj)
+        eq_(res.status_code, 422, res.content)
+        eq_(obj.reget().active, True)
+
+        self.mocks['sub'].cancel.assert_called_with(obj.provider_id)
+
+    def test_active_post(self):
```

Comment:
This would be better named as ``test_cannot_cancel_inactive()`` otherwise it needs a comment or something